sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,171 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+from typing import Callable
+import uuid
+import asyncio
+import logging
+
+from azure.eventhub import EventPosition, EventHubError
+from azure.eventhub.aio import EventHubClient
+from ._cancellation_token import CancellationToken
+from .checkpoint_manager import CheckpointManager
+from .partition_manager import PartitionManager
+from .partition_processor import PartitionProcessor
+from .close_reason import CloseReason
+
+log = logging.getLogger(__name__)
+
+
+class EventProcessor(object):
+    def __init__(self, consumer_group_name: str, eventhub_client: EventHubClient,
+                 partition_processor_callable: Callable[..., PartitionProcessor],
```

Comment:
`partition_processor_callable` should be renamed to something less daunting. Maybe `callback`?

Reply:
I changed to partition_processor_factory for now. Is this a good name?
Other languages use this name.